.ci/azure-pipelines/build-windows.yml

```diff
@@ -27,16 +29,28 @@ jobs:
       - script: |
           vcpkg.exe install eigen3 flann gtest qhull --triplet %PLATFORM%-windows && vcpkg.exe list
         displayName: 'Install Dependencies'
+      - script: |
+          curl -L https://github.com/frerich/clcache/releases/download/v4.2.0/clcache.4.2.0.nupkg --output clcache.4.2.0.nupkg
+          choco install clcache --source=.
+      - task: CacheBeta@0
+        inputs:
+          key: ccache | msvc | $(Agent.OS)
```

Comment:
Shouldn't we have different keys for x86 and x64 jobs? Or is it okay to mix their caches?

Reply:
I thought the platform is different so would the `(Agent.OS)` string, but that's not what happened.

Reply:
We have a step that prints all environment variables. Looking at it, I think it would be appropriate to append `Agent.OSArchitecture` to the key.

Reply:
Oh, and no, it's no ok. Not an issue since the cache is fortunately for reasons unknown to me  empty. Maybe clcache doesn't want the command to include the compiler .... 🤷‍♂